modules/apps/site-initializer/site-initializer-testray/site-initializer-testray-test/src/testFunctional/tests/TestrayBuildPermissions.testcase

```diff
@@ -0,0 +1,341 @@
+		}
+
+		task ("Then can not see the 'Edit, Delete, Promote, Archive' options in kebab") {
+			AssertElementNotPresent(
+				key_kebabOption = "Edit",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+
+			AssertElementNotPresent(
+				key_kebabOption = "Delete",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+
+			AssertElementNotPresent(
+				key_kebabOption = "Promote",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+
+			AssertElementNotPresent(
+				key_kebabOption = "Archive",
+				locator1 = "ObjectAdmin#KEBAB_MENU_OPTION");
+		}
+	}
+
+	@description = "Story ID: LPS-178254 | Test Summary: Verify that the User with the 'Testray Lead' role should be able to create/edit/delete the build"
+	@priority = 4
+	test AssertLeadRoleCanCreateEditAndDelete {
+		property test.name.skip.portal.instance = "TestrayBuildPermissions#AssertLeadRoleCanCreateEditAndDelete";
+
+		task ("Given that the user login with the Testray Lead role") {
+			User.logoutAndLoginPG(
+				userLoginEmailAddress = "lead@testray.com",
+				userLoginFullName = "Lead User");
+
+			Navigator.openSiteURL(siteName = "Testray 2");
+		}
+
+		task ("And the user go to the Build page") {
+			Testray.changePagination(
+				numberItems = 20,
+				valuePagination = 150);
+
+			Testray.goToProjectPage(projectName = ${projectName});
+
+			TestrayRoutines.goToRoutine(routineName = ${routineName});
+		}
+
+		task ("And the user create a build") {
+			TestrayBuild.addBuildViaUI(
+				buildFieldName = "Build 2",
+				caseName = ${caseName},
+				dropdownItem = "New Build",
+				productVersionOption = "1.2",
+				routine = ${routineName});
+		}
+
+		task ("And can see the build in the list") {
+			AssertElementPresent(
+				key_projectName = "Build 2",
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("And the user edit a build") {
+			TestrayContextMenu.clickContextMenuProjects(name = "Build 2");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Edit");
+
+			TestrayBuild.editBuild(
+				description = "Description Edited",
+				fieldName = "Build Edited",
+				gitHash = 12345,
+				productVersion = "1.2",
+				routine = ${routineName});
+		}
+
+		task ("And the build should appear on the list with name edited") {
+			AssertElementPresent(
+				key_projectName = "Build Edited",
+				locator1 = "Testray#ITEM_NAME");
+		}
+
+		task ("When a build is deleted") {
+			TestrayContextMenu.clickContextMenuProjects(name = "Build Edited");
+
+			TestrayContextMenu.selectItemFromProjectsContextMenu(item = "Delete");
+
+			AssertConfirm(value1 = "Are you sure you want to delete this item?");
+		}
+
+		task ("Then the build should not appear on the list") {
+			AssertElementNotPresent(
+				key_projectName = "Build Edited",
+				locator1 = "Testray#ITEM_NAME");
+		}
+	}
+
+	@description = "Story ID: LPS-178254 | Test Summary: Verify that the User with the 'Testray User' role should not be able to create/edit/delete the build but can view"
```

Comment:
One test one intention. Please can you break this test down into 3?

Reply:
Done